pkg/storage/tsdb/config.go

```diff
@@ -151,6 +151,9 @@ type TSDBConfig struct {
 
 	// Positive value enables experiemental support for exemplars. 0 or less to disable.
 	MaxExemplars int `yaml:"max_exemplars"`
+
+	// Enable snapshotting of in-memory TSDB data on disk when shutting down.
+	MemorySnapshotOnShutdown bool `yaml:"memory_snapshot_on_shutdown"`
```

Comment:
We don't need this anymore if we go with per user config.

Reply:
Well, I keep this because the behavior of the snapshot itself is related to the Ingester instance, and we need to have a clearer and more concise switch.

In fact, I still retain the idea of not opening the switch at the tenant level (in the scenario I encountered, the tenant does not actually care about the behavior of the cluster restart).

Reply:
OK, if you don't want to do make the configuration at tenant level then you should remove the limit you add.
We can still discuss with the community to see whether we should make it tenant level or global level. But either way, tenant level should cover the global level already since you can also configure it globally.

But since you added the limit already so I thought you agree with that direction. If the limit is there then we should remove the flag you specify here.

Reply:
We are a SaaS-like service provider, and users do not perceive the existence of any Cortex clusters (does not even support Cortex or Prometheus at the bottom), so in my opinion we only need instance-level configuration. Of course, for me, this is not necessary. :)

Reply:
From another perspective, to some extent _MemorySnapshotOnShutdown_ is similar to the existing _CloseIdleTSDBInterval_.

Reply:
Cortex has different use cases. And as I mentioned multiple times, making it a per tenant limit should cover your use cases, too as you can just set its value to true to make it global.
But I agree, maybe a global value is fine. Let's hear from more people :)

Reply:
I agree with @yeya24 

This option is applied at the start time, and can't be changed at runtime, so I think it should be global only.

Reply:
I see, that's a good point. If it cannot be changed during run time then let's make it global. @wgliang Let's keep the flag only and remove the limit then.